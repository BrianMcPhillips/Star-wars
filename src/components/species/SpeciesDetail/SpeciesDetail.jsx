import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const SpeciesDetail = ({ 
  name,
  classification,
  designation,
  average_lifespan,
  homeworld,
  language,
  people,
  films
}) => (
  <div>
    <h1>{name}</h1>
    <h4>Classification: {classification}</h4>
    <h4>Designation: {designation}</h4>
    <h4>Average Lifespan: {average_lifespan}</h4>
    <h4>Homeworld: {homeworld}</h4>
    <h4>Language: {language}</h4>
    <h4>Characters: {people.map((person, i) => 
      <Link 
        to={`/char-detail/${person.split('people/')[1]}`}
        key={person + i}>
        <p>{person}</p>
      </Link>
    )}</h4>
    <h4>Films: {films.map((film, i) => 
      <Link
        to={`/film-detail/${film.split('films/')[1]}`}
        key={film + i}>
        <p>{film}</p>
      </Link>
    )}</h4>
  </div>
);

SpeciesDetail.propTypes = {
  name: PropTypes.string.isRequired,
  classification: PropTypes.string.isRequired,
  designation: PropTypes.string.isRequired,
  average_lifespan: PropTypes.string.isRequired,
  homeworld: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  people: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  films: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default SpeciesDetail;
