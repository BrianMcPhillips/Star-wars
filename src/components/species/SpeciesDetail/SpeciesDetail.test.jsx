import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SpeciesDetail from './SpeciesDetail';
import { MemoryRouter } from 'react-router-dom';

describe('SpeciesDetail component', () => {
  afterEach(() => cleanup());
  it('renders SpeciesDetail', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SpeciesDetail 
          name="Human"
          classification="mammal"
          designation= "sentient"
          average_lifespan="120"
          homeworld="https://swapi.dev/api/planets/9"
          language="Galactic Basic"
          people={[
            'https://swapi.dev/api/people/12/',
            'https://swapi.dev/api/people/22/'
          ]}
          films={[
            'https://swapi.dev/api/films/1/',
            'https://swapi.dev/api/films/2/',
            'https://swapi.dev/api/films/3/',
            'https://swapi.dev/api/films/6/'
          ]}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
