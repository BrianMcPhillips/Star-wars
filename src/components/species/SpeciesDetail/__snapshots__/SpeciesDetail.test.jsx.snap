// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeciesDetail component renders SpeciesDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Human
    </h1>
    <h4>
      Classification: mammal
    </h4>
    <h4>
      Designation: sentient
    </h4>
    <h4>
      Average Lifespan: 120
    </h4>
    <h4>
      Homeworld: https://swapi.dev/api/planets/9
    </h4>
    <h4>
      Language: Galactic Basic
    </h4>
    <h4>
      Characters: 
      <a
        href="/char-detail/12/"
      >
        <p>
          https://swapi.dev/api/people/12/
        </p>
      </a>
      <a
        href="/char-detail/22/"
      >
        <p>
          https://swapi.dev/api/people/22/
        </p>
      </a>
    </h4>
    <h4>
      Films: 
      <a
        href="/film-detail/1/"
      />
      <a
        href="/film-detail/2/"
      />
      <a
        href="/film-detail/3/"
      />
      <a
        href="/film-detail/6/"
      />
    </h4>
  </div>
</DocumentFragment>
`;
